### PR TITLE
ddl: fix time column accuracy in alter table

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23784

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
fix a bug that time column default is not correct.
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -1727,6 +1727,18 @@ func (s *testIntegrationSuite3) TestAlterColumn(c *C) {
...
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("create table t1(id int)")
+	tk.MustExec("insert into t1(id) values (1);")
+	tk.MustExec("alter table t1 add column update_time datetime NOT NULL DEFAULT CURRENT_TIMESTAMP();")
+	tk.MustExec("alter table t1 add column update_time3 datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3);")
+	tk.MustExec("alter table t1 add column update_time6 datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6);")
+	rows := tk.MustQuery("select * from t1").Rows()
+	c.Assert(rows[0][0], Equals, "1")
+	c.Assert(len(rows[0][1].(string)), Equals, 19)
+	c.Assert(len(rows[0][2].(string)), Equals, 23)
```
Comment (zimulala) [inline ddl/db_integration_test.go:1740]: Before this  PR the result is `2021-08-13 13:45:12.000`, the length feels the same before and after PR, so what is the point of this test?

Comment (xiongjiwei) [reply ddl/db_integration_test.go:1740]: ah.... you're right, this test tests nothing, try to fix.

Comment (xhebox) [reply ddl/db_integration_test.go]: What about a fixed manual default value?

**Review by xhebox — APPROVED** [review]

**Pull request merged**